src/ModelContextProtocol/Auth/OAuthAuthorizationHelpers.cs

```diff
@@ -0,0 +1,268 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using ModelContextProtocol.Auth.Types;
+using ModelContextProtocol.Utils.Json;
+
+namespace ModelContextProtocol.Auth;
+
+/// <summary>
+/// Provides helper methods for handling OAuth authorization.
+/// </summary>
+public static class OAuthAuthorizationHelpers
+{
+    private static readonly HttpClient _httpClient = new();
```

Comment:
This can lead to outages. IHttpClientFactory needs to be used to follow best practices. Sockets sometimes get into a bad state and need to be recycled. That's why periodic recycling is handled by the http client factory internally.